src/shared/components/mutationTable/column/HgvsgColumnFormatter.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from 'react';
+import {Circle} from "better-react-spinkit";
+import 'rc-tooltip/assets/bootstrap_white.css';
+import {Mutation} from "shared/api/generated/CBioPortalAPI";
+import {
+    TableCellStatusIndicator,
+    TableCellStatus,
+    VariantAnnotation
+} from "cbioportal-frontend-commons";
+import GenomeNexusCache, { GenomeNexusCacheDataType } from "shared/cache/GenomeNexusCache";
+
+import hgvsgStyles from "./hgvsg.module.scss";
+
+
+export default class HgvsgColumnFormatter {
+    
+    public static renderFunction(data:Mutation[],
+                                 genomeNexusCache:GenomeNexusCache|undefined) {
+        const genomeNexusCacheData = HgvsgColumnFormatter.getGenomeNexusDataFromCache(data, genomeNexusCache);
+        
+        return <span className={hgvsgStyles["hgvsg-data"]}>{HgvsgColumnFormatter.getHgvsgDataViz(genomeNexusCacheData)}</span>;
+    }
+
+    private static getGenomeNexusDataFromCache(data:Mutation[], cache:GenomeNexusCache|undefined):GenomeNexusCacheDataType | null {
+        if (data.length === 0 || !cache) {
+            return null;
+        }
+        return cache.get(data[0]);
+    }
+
+    private static getHgvsgDataViz(genomeNexusCacheData:GenomeNexusCacheDataType|null) {
+        let status:TableCellStatus | null = null;
+        if (genomeNexusCacheData === null) {
+            status = TableCellStatus.LOADING;
+        } else if (genomeNexusCacheData.status === "error") {
+            status = TableCellStatus.ERROR;
+        } else if (genomeNexusCacheData.data === null) {
+            status = TableCellStatus.NA;
+        } else {
+            let hgvsgData = HgvsgColumnFormatter.getData(genomeNexusCacheData.data);
+            if (hgvsgData == null) {
+                return null;
+            }
+            else {
+                return (
+                    <a href={`https://www.genomenexus.org/variant/${hgvsgData}`} target="_blank" rel="noopener noreferrer">
+                        {hgvsgData}
+                        <img
+                            height='16'
+                            width='16'
+                            src={require("./GenomeNexusLogo.png")}
+                            alt='Genome Nexus'
+                        />
+                    </a>
+                );
+            }        
+        }
+
+        if (status !== null) {
+            // show loading circle
+            if (status === TableCellStatus.LOADING) {
```

Comment:
it's weird that we have a separate case for loading.  one would think that <TableCellStatusIndicator> would take care of loading state.

Reply:
https://github.com/cBioPortal/cbioportal/issues/6988
issue created about the loading state